wolfcrypt/src/port/maxim/max3266x.c

```diff
@@ -0,0 +1,921 @@
+                    return BAD_FUNC_ARG;
+            }
+            return 0; /* True */
+        }
+        else{
+            return 1; /* False */
+        }
+    }
+
+#if !defined(NO_SHA256)
+
+
+
+    WOLFSSL_API int wc_InitSha256_ex(wc_Sha256* sha256, void* heap, int devId)
+    {
+        if (sha256 == NULL){
+            return 1;
+        }
+        (void)heap;
+        (void)devId;
+        return wc_MXC_TPU_SHA_Init((wc_MXC_Sha *)sha256);
+    }
+
+    WOLFSSL_API int wc_InitSha256(wc_Sha256* sha256)
+    {
+        return wc_InitSha256_ex(sha256, NULL, INVALID_DEVID);
+    }
+
+    WOLFSSL_API int wc_Sha256Update(wc_Sha256* sha256, const unsigned char* data, 
+                                        unsigned int len)
+    {
+        return wc_MXC_TPU_SHA_Update(sha256, data, len);
+    }
+
+    WOLFSSL_API int wc_Sha256Final(wc_Sha256* sha256, unsigned char* hash)
+    {
+        return wc_MXC_TPU_SHA_Final((wc_MXC_Sha *)sha256, hash, 
+                                        MXC_TPU_HASH_SHA256);
+    }
+    
+    WOLFSSL_API int wc_Sha256GetHash(wc_Sha256* sha256, unsigned char* hash)
+    {
+        return wc_MXC_TPU_SHA_GetHash((wc_MXC_Sha *)sha256, hash,
+                                        MXC_TPU_HASH_SHA256);
+    }
+
+    WOLFSSL_API int wc_Sha256Copy(wc_Sha256* src, wc_Sha256* dst)
+    {
+        return wc_MXC_TPU_SHA_Copy((wc_MXC_Sha *)src, (wc_MXC_Sha *)dst);
+    }
+
+    WOLFSSL_API void wc_Sha256Free(wc_Sha256* sha256)
+    {
+        wc_MXC_TPU_SHA_Free((wc_MXC_Sha *)sha256);
+        return;
+    }
+
+#endif
+
+
+
+#endif /* MAX3266X_SHA */
+
+#if defined(MAX3266X_MATH)
+
+    int wc_MXC_MAA_init(unsigned int len)
+    {
+        int status;
+        WOLFSSL_MSG("Setting Hardware Mutex and Starting MAA");
+        status = wolfSSL_CryptHwMutexLock();
+        if (status != EXIT_SUCCESS){
+            return status;
+        }
+        status = MXC_TPU_MAA_Init(len);
+        return wc_MXC_error(&status); /* Return Status of Init */
+    }
+
+    int wc_MXC_MAA_Shutdown(void)
+    {
+        int status;
+        WOLFSSL_MSG("Unlocking Hardware Mutex and Shutting Down MAA");
+        status = MXC_TPU_MAA_Shutdown();
+        if (status == E_BAD_PARAM){ /* Miss leading, Send WC_HW_ERROR */
+                                    /* This is returned when MAA cannot stop */
+            return WC_HW_E;
+        }
+        else if(wc_MXC_error(&status) != EXIT_SUCCESS) {
+                return status;
+        }
+        wolfSSL_CryptHwMutexUnLock();
+        return status;
+    }
+
+    /* Update used number for mp_int struct for results */
+    int wc_MXC_MAA_adjustUsed(int *array, unsigned int length)
+    {
+        int lastNonZeroIndex = -1; /* Track the last non-zero index */
+        for (int i = 0; i < length; i++) {
+            if (array[i] != 0) {
+                lastNonZeroIndex = i;
+            }
+        }
+        return (lastNonZeroIndex + 1);
+    }
+
+    void printMAA(mp_int* var, unsigned int len)
```

Comment:
add `static`. Please make this conditional based on build macro for reduced code size. We cannot guarantee printf is available.